File: androidx/src/main/java/me/panpf/androidx/content/ClipIntent.java
package me.panpf.androidx.content;

import android.content.ClipData;
import android.content.ClipDescription;
import android.content.Intent;

import org.jetbrains.annotations.NotNull;

public class ClipIntent extends ClipContent {

    @NotNull
    public Intent intent;

    ClipIntent(@NotNull String mimeType, @NotNull Intent intent) {
        super(mimeType);
        this.intent = intent;
    }

    public ClipIntent(@NotNull Intent intent) {
        this(ClipDescription.MIMETYPE_TEXT_INTENT, intent);
    }

    @Override
    public ClipData.Item toItem() {
        return new ClipData.Item(intent);
    }
}

File: androidx-kt/src/main/java/me/panpf/androidxkt/content/ClipIntent.kt
package me.panpf.androidxkt.content

import android.content.ClipData
import android.content.ClipDescription
import android.content.Intent

class ClipIntent internal constructor(mimeType: String, var intent: Intent) : ClipContent(mimeType) {

    constructor(intent: Intent) : this(ClipDescription.MIMETYPE_TEXT_INTENT, intent) {}

    override fun toItem(): ClipData.Item {
        return ClipData.Item(intent)
    }
}
